tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -760,13 +760,15 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
             if (!kind.isClass) {
               reportError(s"Cannot select $item of non-class $cls")
             } else {
-              maybeClass.foreach {
-                _.lookupField(item).fold[Unit] {
-                  reportError(s"Class $cls does not have a field $item")
-                } { fieldDef =>
-                  if (fieldDef.tpe != tree.tpe)
-                    reportError(s"Select $cls.$item of type "+
-                      s"${fieldDef.tpe} typed as ${tree.tpe}")
+              maybeClass.foreach { c =>
+                if (c.hasInstances) { // see #3060
```

Comment:
Consider moving the reason why we do this inline here. IMO a better position than somewhere in an issue history.

This also makes me thing: Do we need to support a "full check mode" where essentially the `BaseLinker` doesn't remove stuff (or at least does not tear classes apart)? If I were to work on an IR patcher, I would be terribly annoyed by behavior like this.